### ISSUE TITLE
Basic auth support for client

### ISSUE THREAD

**Issue by thekad** [RobustPerception/PushProx#12]
Hi,

I have a use case that I believe would be fairly common for PushProx: My prometheus server is behind an https+basicauth endpoint over the internet, but the client part of PushProx doesn't have support for user/pass authentication. Is this something that would make sense to add to PushProx?

**Comment by brian-brazil** [comment]
The security setup on the Prometheus isn't relevant, neither the PushProx nor the client initiate connections to Prometheus.

**Comment by thekad** [comment]
I'm sorry I wasn't clear: I meant the server part of PushProx, not prometheus. Basically I have satellite servers that can't talk directly to prometheus, it would make sense to put PushProx in a DMZ-like place where both clients and prometheus can reach, I am basically asking about this from the README: "There is no authentication or authorisation included, a reverse proxy can be put in front though to add these."
In other words: having a reverse proxy on the server part so I can protect it with user and password, how do I make the client part send a user and password?

**Comment by brian-brazil** [comment]
You should be able to use basic auth via the URL, with the usual caveats.

**Comment by thekad** [comment]
I see, thanks!
